workflows/differentialabundance.nf

```diff
@@ -193,6 +195,15 @@ workflow DIFFERENTIALABUNDANCE {
             }
             .flatten()
             .unique() // Uniquify to keep each contrast variable only once (in case it exists in multiple lines for blocking etc.)
+
+        VALIDATE_YML_MODEL (
```

Comment:
Sorry, one last thought. 

The issue with doing it this way is that `ch_input` etc will proceed through other processes even if there are problems that eventually get flagged by this process. What we really want is for this to stop things before they get that far.

You can use a dummy join to make that happen- I do that in [this subworkflow](https://github.com/nf-core/modules/blob/4d95826db394fc94c9e9ea11170bbe19e0d849ca/subworkflows/nf-core/fastq_qc_trim_filter_setstrandedness/main.nf#L156) for example. Or you can just have VALIDATE_YML_MODEL spit the matrix and contrasts back out again.

Reply:
Work was already done to enable yaml contrasts there: https://github.com/pinin4fjords/shinyngs/pull/68